lib/chat/key_share.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Chat.KeyShare do
+  @moduledoc "Manipulate social sharing keys"
+
+  alias Chat.{Dialogs, Dialogs.Dialog, Identity}
+  alias Chat.{ChunkedFiles, ChunkedFilesMultisecret}
+  alias Chat.Upload.UploadKey
+
+  @threshold 4
+
+  def generate_key_shares({%Identity{private_key: private_key} = _me, users}) do
+    shares =
+      private_key
+      |> Enigma.hide_secret_in_shares(users |> Enum.count(), @threshold)
+
+    share_count = Enum.count(shares)
+
+    for i <- 0..(share_count - 1), into: [] do
+      %{
+        user: Enum.at(users, i),
+        key: Enum.at(shares, i) |> Base.encode64()
+      }
+    end
```

Comment:
almost Enum.zip

Reply:
done using Enum.zip_reduce/4 instead of for